app/scripts/app-init.js

```diff
@@ -184,11 +184,22 @@ const registerInPageContentScript = async () => {
   }
 };
 
+// On MV3 builds we must listen for this event in `app-init`, otherwise we found that the listener
+// is never called.
+// For MV2 builds, the listener is added in `background.js` instead.
 chrome.runtime.onInstalled.addListener(function (details) {
   if (details.reason === 'install') {
-    chrome.storage.session.set({ isFirstTimeInstall: true });
-  } else if (details.reason === 'update') {
-    chrome.storage.session.set({ isFirstTimeInstall: false });
+    // This condition is for when `background.js` was loaded before the `onInstalled` listener was
+    // called.
+    if (globalThis.__metamaskTriggerOnInstall) {
```

Comment:
do you think adding types to our `types/global.d.ts` (or even a new `types/*.d.ts` file) is worthwhile for these global vars?

Reply:
Couldn't hurt!

Reply:
Done here: https://github.com/MetaMask/metamask-extension/pull/31332/commits/8d0dbcd3a2df3815837fe3c6cc218f89de6e45e9